src/Icicle/Avalanche/Statement/Simp/Constructor.hs

```diff
@@ -211,17 +217,24 @@ constructor a_fresh statements
                                           (primUnpack ix         tx  nv)) tis
 
    -- comparison
-   | (PrimMinimal (Min.PrimRelation Min.PrimRelationEq t), [nx, ny]) <- prima
+   | (PrimMinimal (Min.PrimRelation primb t), [nx, ny]) <- prima
    , Just ts <- tryMeltType t
    , tis     <- List.zip ts [0..]
+   , primc   <- relation primb
    = primFold1 primAnd
-   $ fmap (\(tv, i) -> primEq tv (primUnpack i t nx)
-                                 (primUnpack i t ny)) tis
+   $ fmap (\(tv, i) -> primc tv (primUnpack i t nx)
+                                (primUnpack i t ny)) tis
 
 
    | otherwise
    = x
 
+  relation Min.PrimRelationGt = primGt
+  relation Min.PrimRelationGe = primGe
+  relation Min.PrimRelationLt = primLt
+  relation Min.PrimRelationLe = primLe
+  relation Min.PrimRelationEq = primEq
+  relation Min.PrimRelationNe = primNe
```

Comment:
it's more annoying than it looks! less than is something like..

```
(a,b) < (c,d)
==>
(a < c) || (a == c && b < d)
```
I think.
I don't think ord is particularly important but if /= is easy to fix might as well do that